src/Mauro/Greeter.test.ts

```diff
@@ -36,4 +36,17 @@ describe("Greeter.greet", () => {
     // assert
     expect(result).toBe("Good night Mauro");
   });
+
+  it("should log into console when function its called", () => {
+    // arrange
+    const currentTime = new Date("2022-05-08T05:59:00");
+    const greeter = new Greeter(currentTime);
+    console.log = jest.fn();
```

Comment:
Here, the function `log` in the global object `window.console` is being modified.

I am not sure, but could it affect other tests? should you clean up this setup after the test?

On the other hand, one of the supposed benefits of the TDD is that it helps you to define systems with clear explicit dependencies. What do you think about injecting the log function in the greeter constructor?